backend/Utils/JWTUtils.ts

```diff
@@ -1,10 +1,24 @@
 import jwt from "jsonwebtoken";
+import { JWTModel } from "../DB/models/JWT";
 
 const { sign, verify } = jwt;
 
 export function makeJWT(payload: object): string {
   const secret = process.env.JWT_PASSWORD_RESET_SECRET || "secret";
 
   const token = sign(payload, secret, { expiresIn: "1h" });
+
+  const userId = payload["userId"];
+
+  saveJWT(token, userId);
```

Comment:
Like I mentioned in my other comment, I think JWTModel.create() should be in ResetPassword.ts, not JWTUtils, so this should be removed. However, if it were not being removed there is a bug here because we're not awaiting saveJWT. Since saveJWT is an async function, it returns a promise. Returning the promise happens synchronously but that doesn't guarantee that the asynchronous call to the database will succeed - for that you have to check whether the promise resolves or rejects. Using the `await` keyword stops the rest of the function from executing until the promise resolves, which is what we'd want here so that if the JWT doesn't save the makeJWT function's returned promise rejects. So it would be `await saveJWT(token, userId)` to fix it (but we don't want to do that here since this should be removed)

Promises are something often glossed over in bootcamps, I'm happy to go over them in more detail with you if that would help.

Reply:
Thank you! Very helpful to review promises.